addons/portal/controllers/thread.py

```diff
@@ -0,0 +1,37 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import http
+from odoo.tools import html2plaintext
+from odoo.http import request
+from odoo.addons.mail.controllers import thread
+
+
+class ThreadController(thread.ThreadController):
+
+    @http.route()
+    def mail_message_post(self, thread_model, thread_id, post_data, context=None, **kwargs):
+        message = super().mail_message_post(thread_model, thread_id, post_data, context, **kwargs)
+        message_data = message.get("mail.message")
+        if (message_data := message_data and message_data[0]) and request.env[
+            "res.partner"
+        ]._get_portal_partner_from_context():
+            message = {
+                "default_message": html2plaintext(message_data.get("body")),
+                "default_message_id": message_data.get("id"),
+                "default_attachment_ids": message.get("ir.attachment"),
+            }
```

Comment:
It's not good to completely change the result value like that. We should get rid of this override now.

Whatever needs this result should be adapted to properly use the standard message result from a `store.insert`.

If not possible or not desirable to have the store there now, at least this transformation should be done in JS rather than in python, so that we don't need to rely on `_get_portal_partner_from_context` to detect where the call is coming from.

I'm trying to remove the need for `_get_portal_partner_from_context` and the need to update context.